interfaces/builtin/zigbee_dongle.go

```diff
@@ -0,0 +1,137 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+// ZigbeeDongleInterface is the type for serial port interfaces.
+type ZigbeeDongleInterface struct{}
+
+// Name of the zigbee-dongle interface.
+func (iface *ZigbeeDongleInterface) Name() string {
+	return "zigbee-dongle"
+}
+
+func (iface *ZigbeeDongleInterface) String() string {
+	return iface.Name()
+}
+
+var zigbeeDongleDevPath = "/dev/zigbee/*"
+
+var zigbeeDonglePermanentSlotUdev = []byte(`
+IMPORT{builtin}="usb_id"
+SUBSYSTEM=="tty", SUBSYSTEMS=="usb", ATTRS{idProduct}=="0003", ATTRS{idVendor}=="10c4", SYMLINK+="zigbee/$env{ID_SERIAL}"
+`)
+
+// SanitizeSlot checks slot validity
+func (iface *ZigbeeDongleInterface) SanitizeSlot(slot *interfaces.Slot) error {
+	// check slot name
+	if iface.Name() != slot.Interface {
+		panic(fmt.Sprintf("slot is not of interface %q", iface))
+	}
+	return nil
+}
+
+// PermanentSlotSnippet - no permissions given to slot permanently
+func (iface *ZigbeeDongleInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityDBus:
+		return nil, nil
+	case interfaces.SecurityUDev:
+		return zigbeeDonglePermanentSlotUdev, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+// ConnectedSlotSnippet - no permissions given to slot on connection
+func (iface *ZigbeeDongleInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor, interfaces.SecurityDBus, interfaces.SecuritySecComp, interfaces.SecurityUDev:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+// SanitizePlug checks plug validity
+func (iface *ZigbeeDongleInterface) SanitizePlug(slot *interfaces.Plug) error {
+	if iface.Name() != slot.Interface {
+		panic(fmt.Sprintf("plug is not of interface %q", iface))
+	}
+	return nil
+}
+
+// PermanentPlugSnippet no permissions provided to plug permanently
+func (iface *ZigbeeDongleInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityDBus, interfaces.SecurityUDev:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+// ConnectedPlugSnippet returns security snippet specific to the plug
+func (iface *ZigbeeDongleInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		paths, err := iface.zigbeeDevPaths(slot)
```

Comment:
Why are we doing this instead of simply using the glob itself in the apparmor profile?